### PR TITLE
K8s worker maintenance

### PR DESCRIPTION
**Pull request by mkalcok**

### CODE REVIEW

File: juju-tutorials/how-to-perform-maintenance-on-a-k8s-worker.md [hunk 1]
```diff
@@ -0,0 +1,178 @@
+[comment]: <> (How To Perform Maintenance On Kubernetes Worker)
```
Comment (auria) [inline juju-tutorials/how-to-perform-maintenance-on-a-k8s-worker.md]: "On A Kubernetes Worker"

File: juju-tutorials/how-to-perform-maintenance-on-a-k8s-worker.md [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+### Environment preparation
+
+In case that you already have a juju model running a kubernetes cluster with
+multiple workers, you can skip this section.
+
+This tutorial can be completed with either [Charmed Kubernetes](https://jaas.ai/canonical-kubernetes)
+or with [Kubernetes Core](https://jaas.ai/kubernetes-core). For testing
+purposes, Kubernetes Core is enough.
+
+Start by deploying Kubernetes Core bundle and increasing number of Kubernetes
```
Comment (auria) [inline juju-tutorials/how-to-perform-maintenance-on-a-k8s-worker.md]: "and increasing the number of Kubernetes"

File: juju-tutorials/how-to-perform-maintenance-on-a-k8s-worker.md [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+kubectl get pod -o wide
+NAME                                READY   STATUS    RESTARTS   AGE   IP            NODE                NOMINATED NODE   READINESS GATES
+nginx-deployment-7848d4b86f-9zcl2   1/1     Running   0          25s   10.1.94.13    juju-37550a-k8s-3   <none>           <none>
+nginx-deployment-7848d4b86f-hpjqv   1/1     Running   0          25s   10.1.100.13   juju-37550a-k8s-1   <none>           <none>
+nginx-deployment-7848d4b86f-pzqkt   1/1     Running   0          25s   10.1.94.14    juju-37550a-k8s-3   <none>           <none>
+nginx-deployment-7848d4b86f-tn7bf   1/1     Running   0          25s   10.1.100.12   juju-37550a-k8s-1   <none>           <none>
+```
+
+[note status="What to do if you get stuck"]
+The best place to ask for help is the [Juju Discourse forum](https://discourse.charmhub.io/).
```
Comment (auria) [inline juju-tutorials/how-to-perform-maintenance-on-a-k8s-worker.md]: "Juju Discourse forum" has been renamed to "Charmhub Discourse forum".

File: juju-tutorials/how-to-perform-maintenance-on-a-k8s-worker.md [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+[/note]
+
+## Draining Kubernetes Worker
+
+Duration: 10:00
+
+The kubernetes-worker charm provides actions for draining workers. These
+actions are pause and resume. Behind the scenes, pause runs a kubectl command
+to drain the node of all pods running on it. These pods will be rescheduled
+onto other nodes.
```
Comment (auria) [inline juju-tutorials/how-to-perform-maintenance-on-a-k8s-worker.md]: Suggestion: put "pause" and "resume" strings between backsticks (to highlight they are actions).

File: juju-tutorials/how-to-perform-maintenance-on-a-k8s-worker.md [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+
+First check to make sure there's enough resource capacity in the cluster to
```
Comment (auria) [inline juju-tutorials/how-to-perform-maintenance-on-a-k8s-worker.md]: "First, make sure" (missing comma and... shorter imperative?)

File: juju-tutorials/how-to-perform-maintenance-on-a-k8s-worker.md [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+
+```console
+$ kubectl top nodes
+NAME                CPU(cores)   CPU%   MEMORY(bytes)   MEMORY%   
+juju-37550a-k8s-1   1121m        28%    1476Mi          21%       
+juju-37550a-k8s-3   1185m        29%    1458Mi          21%
+```
+
+Then you can drain node of your choice identified by unit ID. To drain
+kubernetes-worker/0 run following.
```
Comment (auria) [inline juju-tutorials/how-to-perform-maintenance-on-a-k8s-worker.md]: Backsticks between "kubernetes-worker/0" ?

File: juju-tutorials/how-to-perform-maintenance-on-a-k8s-worker.md [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+Duration: 2:00
+
+Once the worker is ready to be brought back into circulation, you can bring
+it back online by running:
+
+```console
+juju run-action --wait kubernetes-worker/0 resume
+```
+[note status="Rebalancing of the cluster"]
+Pods that were previously migrated from this node wont be automatically
```
Comment (auria) [inline juju-tutorials/how-to-perform-maintenance-on-a-k8s-worker.md]: "won't"

**Review by auria — COMMENTED** [review]
Overall, lgtm. Some suggestions on wording but tutorial can be followed / tested. Thank you!

File: juju-tutorials/how-to-perform-maintenance-on-a-k8s-worker.md (hunk 3 above)
Comment (mkalcok) [reply juju-tutorials/how-to-perform-maintenance-on-a-k8s-worker.md]: I'll change it in the template as well.

**Comment by auria** [comment]
LGTM, thank you.

**Pull request merged**